### ISSUE TITLE
redundant copy

### ISSUE THREAD

**Issue by phoenix20162016** [secretflow/yacl#33]
![image](https://user-images.githubusercontent.com/18574396/222692943-f82583f7-2e77-49b3-829e-04705e33e4e8.png)

**Comment by usafchn** [comment]
hello @phoenix20162016 

Nice catch~

The signature of `SendAsyncImpl` method is:
```
void SendAsyncImpl(const std::string& key, ByteContainerView value)
```

The `ByteContainerView` is a view type that cannot take ownership of Buffer. `ByteContainerView` is essentially a `Span`, which only records the address and length of the `Buffer`, so there is no copy here.

On the other hand, using the `Buffer&&` type as a parameter of the `SendAsync` method is unreasonable. It may be better to change it to a constant reference type. We will re-examine the signature of `SendAsync` and fix it later.

**Comment by phoenix20162016** [comment]
as an async method,  in sometimes, ByteContainerView is not very suitable, cause you must be carefully think about the life time of the source, but right reference buffer is not

**Comment by usafchn** [comment]
> as an async method, in sometimes, ByteContainerView is not very suitable, cause you must be carefully think about the life time of the source, but right reference buffer is not

Yes, you are right, I think this is a serious problem, we will fix it as soon as possible.

**Comment by anakinxc** [comment]
Fixed via #40
